packages/web-app-user-management/src/components/Users/CreateUserModal.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <oc-modal
+    :title="$gettext('Create user')"
+    :button-cancel-text="$gettext('Cancel')"
+    :button-confirm-text="$gettext('Create')"
+    :button-confirm-disabled="buttonConfirmDisabled"
+    @cancel="$emit('cancel')"
+    @confirm="$emit('confirm', user)"
+  >
+    <template #content>
+      <oc-text-input
+        v-model="user.onPremisesSamAccountName"
+        class="oc-mb-s"
+        :label="$gettext('User name') + '*'"
+        :error-message="formData.userName.errorMessage"
+        :fix-message-line="true"
+        @input="validateUserName"
+      />
+      <oc-text-input
+        v-model="user.displayName"
+        class="oc-mb-s"
+        :label="$gettext('First and last name') + '*'"
+        :error-message="formData.displayName.errorMessage"
+        :fix-message-line="true"
+        @input="validateDisplayName"
+      />
+      <oc-text-input
+        v-model="user.mail"
+        class="oc-mb-s"
+        :label="$gettext('Email') + '*'"
+        :error-message="formData.email.errorMessage"
+        type="email"
+        :fix-message-line="true"
+        @change="validateEmail"
+      />
+      <oc-text-input
+        v-model="user.passwordProfile.password"
+        class="oc-mb-s"
+        :label="$gettext('Password') + '*'"
+        :error-message="formData.password.errorMessage"
+        type="password"
+        :fix-message-line="true"
+        @input="validatePassword"
+      />
+    </template>
+  </oc-modal>
+</template>
+
+<script>
+import * as EmailValidator from 'email-validator'
+
+export default {
+  name: 'CreateUserModal',
+  data: function () {
+    return {
+      formData: {
+        userName: {
+          errorMessage: '',
+          valid: false
+        },
+        displayName: {
+          errorMessage: '',
+          valid: false
+        },
+        email: {
+          errorMessage: '',
+          valid: false
+        },
+        password: {
+          errorMessage: '',
+          valid: false
+        }
+      },
+      user: {
+        onPremisesSamAccountName: '',
+        displayName: '',
+        mail: '',
+        passwordProfile: {
+          password: ''
+        }
+      }
+    }
+  },
+  computed: {
+    buttonConfirmDisabled() {
+      let invalid = false
+      Object.keys(this.formData).forEach((key) => {
+        if (this.formData[key].valid === false) {
+          invalid = true
+        }
+      })
+      return invalid
+    }
+  },
+  methods: {
+    validateUserName() {
```

Comment:
the validation logic is the same for all fields, only the key & errorMessage is different, is it possible to use the same method for all validations?